tests/test_http_stats.py

```diff
@@ -0,0 +1,91 @@
+"""Validate that the worker id in the context of the http proxy is the id of the worker rather than the queue."""
+
+import unittest
+from http.server import HTTPServer, BaseHTTPRequestHandler
+
+from saq import Queue, Worker
+from saq.queue.http import HttpProxy
+from saq.types import Context
+from tests.helpers import setup_postgres, create_postgres_queue, teardown_postgres
+import asyncio
+import threading
+
+
+async def echo(_ctx: Context, *, a: int) -> int:
+    return a
+
+
+class ProxyRequestHandler(BaseHTTPRequestHandler):
+    def __init__(self, *args, proxy=None, **kwargs):
+        self.proxy = proxy
+        super().__init__(*args, **kwargs)
+
+    def do_POST(self):
+        length = int(self.headers["Content-Length"])
+        body = self.rfile.read(length).decode("utf-8")
+        response = asyncio.run(self.proxy.process(body))
+        if response:
+            self.send_response(200)
+            self.send_header("Content-type", "application/json")
+            self.end_headers()
+            self.wfile.write(response.encode("utf-8"))
+        else:
+            self.send_response(200)
+            self.end_headers()
+
+
+class TestQueue(unittest.IsolatedAsyncioTestCase):
+    async def asyncSetUp(self) -> None:
+        await setup_postgres()
+
+    async def asyncTearDown(self) -> None:
+        await teardown_postgres()
+
+    async def test_http_proxy_with_two_workers(self) -> None:
+        queue = await create_postgres_queue()
+        proxy = HttpProxy(queue=queue)
+
+        server = HTTPServer(
+            ("localhost", 8080),
+            lambda *args, **kwargs: ProxyRequestHandler(*args, proxy=proxy, **kwargs),
+        )
+        server_thread = threading.Thread(target=server.serve_forever)
```

Comment:
do we do this anywhere else? i think the general pattern is to use asyncio web app right?

Reply:
moved over